packages/firestore/src/local/simple_db.ts

```diff
@@ -299,9 +304,33 @@ export class SimpleDb {
         // https://developer.mozilla.org/en-US/docs/Web/API/IDBVersionChangeRequest/setVersion
         const request = indexedDB.open(this.name, this.version);
 
+        // Store information about "Clear Site Data" being detected in the
+        // "onupgradeneeded" event listener and handle it in the "onsuccess"
+        // event listener, as opposed to throwing directly from the
+        // "onupgradeneeded" event listener. Do this because throwing from the
+        // "onupgradeneeded" event listener results in a generic error being
+        // reported to the "onerror" event listener that cannot be distinguished
+        // from other errors.
+        const clearSiteDataEvent: ClearSiteDataDatabaseDeletedEvent[] = [];
+
         request.onsuccess = (event: Event) => {
+          let error: unknown;
+          if (clearSiteDataEvent[0]) {
```

Comment:
the c++ side of my brain hurts when i see this :) 

TS playground suggests it's safe to do, but why not use `if (clearSiteDataEvent.length > 0) {` ?

Reply:
What, you don't like the looks of accessing uninitialized memory? ;)

My memory told me that the TypeScript compiler was failing to infer that `clearSiteDataEvent[0]` is defined when the `if` statement checked `clearSiteDataEvent.length > 0`; however, I just tested it out and it appears to compile just fine. So, I've changed if to check for a non-zero length.